src/osuTK/Input/JoystickState.cs

```diff
@@ -104,7 +105,19 @@ public JoystickHatState GetHat(JoystickHat hat)
         /// <param name="button">The button to query.</param>
         public bool IsButtonDown(int button)
         {
-            return (buttons & ((long)1 << button)) != 0;
+            if (button < 0 || button >= MaxButtons)
+            {
+                Debug.Print("[Joystick] IsButtonDown on invalid button {0}", button);
+                return false;
+            }
+
+            unsafe
+            {
+                fixed (bool* pbuttons = buttons)
```

Comment:
Why not just `buttons[button]`? perf gain is marginal if any at all.

Reply:
current language level does not allow for  that. i asked in the comments above if it's ok to increase but got no answer. also the axes are written the same way.

Reply:
opentk is silly. the fix is to remove the `unsafe fixed` stuff in the first place.